File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        
        sort(begin(points), end(points));
        
        vector<int> prev = points[0];
        
        int count = 1;
        for(int i=1; i<points.size(); i++){
            
            int prevStartingPoint = prev[0];
            int prevEndingPoint = prev[1];
            
            vector<int> cur = points[i];
            int curStartingPoint = cur[0];
            int curEndingPoint = cur[1];
            
            if(curStartingPoint > prevEndingPoint){
                // no overlap
                prev = cur;
                count++;
            }
            else{
                prev[0] = max(prevStartingPoint, curStartingPoint);
                prev[1] = min(prevEndingPoint, curEndingPoint);
            }
        }
        
        return count;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    
    points.sort(function(a, b){
        return a[0] - b[0];
    });
    
    let prev = points[0]
    
    let count = 1;
    for(let i=1; i<points.length; i++){
        
        let curStartPoint = points[i][0];
        let curEndPoint = points[i][1];
        
        let prevStartPoint = prev[0];
        let prevEndPoint = prev[1];
        
        if(curStartPoint > prevEndPoint){
            // no overlap
            count++;
            prev = points[i];
        }
        else{
            prev[0] = Math.max(prev[0], curStartPoint);
            prev[1] = Math.min(prev[1], curEndPoint);
        }
    }
    
    return count;
};